src/app/model/ChessEngine.ts

```diff
@@ -33,9 +37,22 @@ export class ChessEngine implements ChessModel {
       to: squareTo,
     };
     const pieceWasCaptured = this.pieceWasCaptured(squareTo, chosenPiece);
+    if (pieceWasCaptured) {
+      events.push(pieceWasCaptured);
+    }
 
     this.onPieceWasMoved(pieceWasMoved);
-    return pieceWasCaptured ? [pieceWasMoved, pieceWasCaptured] : [pieceWasMoved];
+    events.push(pieceWasMoved);
+
+    const kingWasChecked = this.kingWasChecked(chosenPiece);
+    if (kingWasChecked) {
```

Comment:
Zobacz jak fajnie Ania dała radę to rozwiązać bez ifów :) 
https://github.com/nowakprojects/CodersCamp2020.Project.TypeScript.Chess/pull/80